acestep/ui/gradio/events/generation/mode_ui_test.py

```
@@ -221,5 +221,52 @@ def test_leaving_extract_sets_auto_fields_non_interactive(self):
+        gradio_temp_path = r"C:\Users\test\AppData\Local\Temp\gradio\abc\song.wav"
+        with patch("soundfile.info", return_value=fake_info) as mock_info:
+            result = handle_extract_src_audio_change(gradio_temp_path, "Extract")
+            mock_info.assert_called_once_with(gradio_temp_path)
+        # Result should carry a positive numeric value without exception.
+        self.assertEqual(getattr(result, "_props", {}).get("value"), 42.0)
+
+    def test_swallows_invalid_audio_errors(self):
+        """A bad/unreadable file should be logged-and-skipped, not raised."""
+        from unittest.mock import patch
+        with patch("soundfile.info", side_effect=RuntimeError("bad file")):
+            result = handle_extract_src_audio_change("/tmp/bogus.wav", "Lego")
```

Ruff (0.15.11)

[error] Probable insecure usage of temporary file or directory: "/tmp/bogus.wav"

(S108)